sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -422,12 +422,30 @@ case class DataSourceStrategy(conf: SQLConf) extends Strategy with Logging with
 }
 
 object DataSourceStrategy {
+
+  /**
+   * Tries to translate a Catalyst [[Expression]] into data source [[Expression]].
+   *
+   * @return a `Some[catalog.v2.expressions.Expression]` if the input [[Expression]]
+   *         is convertible, otherwise a `None`.
+   */
+  protected[sql] def translateExpression(
```

Comment:
I think this should be located with the Expression classes instead of in the v1 DataSource code.

Reply:
Is `Expression` the right place where this util should go?  This util translate the inner `Expression` to the public `catalog.v2.expressions.Expression` and it belongs to nether of the two. My first thinking is there's another `translateFilter` in `DataSourceAnalysis` which translate the inner `Expression` to the public datasource `Filter`, it has similar function with `translateExpression`.

Reply:
can we create a new object in `spark.sql.connector.expression` to put this method?

Reply:
Sorry @cloud-fan , what does the package `spark.sql.connector` mean? I see `InMemoryTable` was moved there, and this package only exists in test codes.

Reply:
how about the `LogicalExpressions` object?

Reply:
`LogicalExpressions` seems ok. Moved the util function there.